### PR TITLE
develop/move sel to workspace and modify tests to support it

### PR DESCRIPTION
**Pull request by WinnieXU1**
1.move /var/tmp/ipmi_sim to ~/.infrasim/<node_name>/data/ipmi_sim
2.modify some tests to support it

### CODE REVIEW

**Comment by emcinfrasim** [comment]
```
******* Unit Test ******* 
TestName: test_ipmi_process_start 
ErrorType: exceptions.AssertionError 
ErrorMessage:  
```

**Comment by emcinfrasim** [comment]
```
******* Functional Test ******* 
TestName: test_qemu_local_sel 
ErrorType: exceptions.AssertionError 
ErrorMessage: 
-------------------- >> begin captured stdout << ---------------------
   1 |  Pre-Init  |0002876441| Power Unit #0x02 | Redundancy Lost | Asserted
   2 |  Pre-Init  |0002876441| Power Unit #0x02 | Non-Redundant: Sufficient from Redundant | Asserted
   3 |  Pre-Init  |0002876441| Module/Board #0x0e | Device Present | Asserted
   4 |  Pre-Init  |0002876441| Power Supply #0x50 | Presence detected | Asserted
   5 |  Pre-Init  |0002876441| Power Supply #0x50 | Power Supply AC lost | Asserted
   6 |  Pre-Init  |0002876441| Power Supply #0x51 | Presence detected | Asserted
   7 |  Pre-Init  |0002876441| Processor #0x70 | Presence detected | Asserted
   8 |  Pre-Init  |0002876441| Processor #0x71 | Presence detected | Asserted
   9 |  Pre-Init  |0002876441| Drive Slot #0xf0 | Drive Present | Asserted
   a |  Pre-Init  |0002876441| Drive Slot #0xf1 | Drive Present | Asserted
   b |  Pre-Init  |0002876441| Drive Slot #0xf2 | Drive Present | Asserted


--------------------- >> end captured stdout << ----------------------
-------------------- >> begin captured logging << --------------------
paramiko.transport: DEBUG: starting thread (client mode): 0x5d017490L
paramiko.transport: DEBUG: Local version/idstring: SSH-2.0-paramiko_2.0.1
paramiko.transport: DEBUG: Remote version/idstring: SSH-2.0-OpenSSH_5.4
paramiko.transport: INFO: Connected (version 2.0, client OpenSSH_5.4)
paramiko.transport: DEBUG: kex algos:[u'diffie-hellman-group-exchange-sha256', u'diffie-hellman-group-exchange-sha1', u'diffie-hellman-group14-sha1', u'diffie-hellman-group1-sha1'] server key:[u'ssh-rsa', u'ssh-dss'] client encrypt:[u'aes128-ctr', u'aes192-ctr', u'aes256-ctr', u'arcfour256', u'arcfour128', u'aes128-cbc', u'3des-cbc', u'blowfish-cbc', u'cast128-cbc', u'aes192-cbc', u'aes256-cbc', u'arcfour', u'rijndael-cbc@lysator.liu.se'] server encrypt:[u'aes128-ctr', u'aes192-ctr', u'aes256-ctr', u'arcfour256', u'arcfour128', u'aes128-cbc', u'3des-cbc', u'blowfish-cbc', u'cast128-cbc', u'aes192-cbc', u'aes256-cbc', u'arcfour', u'rijndael-cbc@lysator.liu.se'] client mac:[u'hmac-md5', u'hmac-sha1', u'umac-64@openssh.com', u'hmac-ripemd160', u'hmac-ripemd160@openssh.com', u'hmac-sha1-96', u'hmac-md5-96'] server mac:[u'hmac-md5', u'hmac-sha1', u'umac-64@openssh.com', u'hmac-ripemd160', u'hmac-ripemd160@openssh.com', u'hmac-sha1-96', u'hmac-md5-96'] client compress:[u'none', u'zlib@openssh.com'] server compress:[u'none', u'zlib@openssh.com'] client lang:[u''] server lang:[u''] kex follows?False
paramiko.transport: DEBUG: Kex agreed: diffie-hellman-group1-sha1
paramiko.transport: DEBUG: Cipher agreed: aes128-ctr
paramiko.transport: DEBUG: MAC agreed: hmac-md5
paramiko.transport: DEBUG: Compression agreed: none
paramiko.transport: DEBUG: kex engine KexGroup1 specified hash_algo <built-in function openssl_sha1>
paramiko.transport: DEBUG: Switch to new keys ...
paramiko.transport: DEBUG: Adding ssh-rsa host key for [127.0.0.1]:2222: 50101a92e2522e3d71fb2ee34ded20cd
paramiko.transport: DEBUG: Trying discovered key c3b3c5b914bd1e0426054008b3f50ebd in /home/infrasim/.ssh/id_rsa
paramiko.transport: DEBUG: userauth is OK
paramiko.transport: INFO: Authentication (publickey) failed.
paramiko.transport: DEBUG: userauth is OK
paramiko.transport: INFO: Authentication (password) successful!
paramiko.transport: DEBUG: [chan 0] Max packet in: 32768 bytes
paramiko.transport: DEBUG: [chan 0] Max packet out: 32768 bytes
paramiko.transport: DEBUG: Secsh channel 0 opened.
paramiko.transport: DEBUG: [chan 0] Sesch channel 0 request ok
paramiko.transport: DEBUG: [chan 1] Max packet in: 32768 bytes
paramiko.transport: DEBUG: [chan 1] Max packet out: 32768 bytes
paramiko.transport: DEBUG: Secsh channel 1 opened.
paramiko.transport: DEBUG: [chan 0] EOF sent (0)
paramiko.transport: DEBUG: [chan 1] Sesch channel 1 request ok
paramiko.transport: DEBUG: Ignoring message for dead channel 0
paramiko.transport: DEBUG: Ignoring message for dead channel 0
paramiko.transport: DEBUG: Ignoring message for dead channel 0
paramiko.transport: DEBUG: [chan 1] EOF received (1)
--------------------- >> end captured logging << --------------------- 
```

**Comment by emcinfrasim** [comment]
```
******* Unit Test ******* 
TestName: test_ipmi_process_start 
ErrorType: exceptions.AssertionError 
ErrorMessage:  
```

File: infrasim/model.py [hunk 1]
```diff
@@ -1772,8 +1772,9 @@ def init(self):
 
 
     def get_commandline(self):
-        ipmi_cmd_str = "{0} -c {1} -f {2} -n -s /var/tmp".\
-            format(self.__bin, self.__config_file, self.__emu_file)
+        path = os.path.join(config.infrasim_home, self.__node_name, "data")
+        ipmi_cmd_str = "{0} -c {1} -f {2} -n -s {3}" .\
+            format(self.__bin, self.__config_file, self.__emu_file, path)
```
Comment (turtle-fly) [inline infrasim/model.py]: CBMC can `get_workspace()`, so you don't have to compose path from `config.infrasim_home` and `self.__node_name`, which means, you don't have to change `infrasim/ipmi.py`

(I sincerely wan't to remove `infrasim/ipmi.py` someday...)

**Comment by turtle-fly** [comment]
Good to go.

**Pull request merged**